source/client/output_formatter_impl.cc

```diff
@@ -138,5 +144,123 @@ DottedStringOutputFormatterImpl::formatProto(const nighthawk::client::Output& ou
   return ss.str();
 }
 
+const nighthawk::client::Result&
+FortioOutputFormatterImpl::getGlobalResult(const nighthawk::client::Output& output) const {
+  for (const auto& nh_result : output.results()) {
+    if (nh_result.name() == "global") {
+      return nh_result;
+    }
+  }
+
+  throw NighthawkException("Nighthawk output was malformed, contains no 'global' results.");
+}
+
+const nighthawk::client::Counter&
+FortioOutputFormatterImpl::getCounterByName(const nighthawk::client::Result& result,
+                                            absl::string_view counter_name) const {
+  for (const auto& nh_counter : result.counters()) {
+    if (nh_counter.name() == counter_name) {
+      return nh_counter;
+    }
+  }
+
+  throw NighthawkException(absl::StrCat(
+      "Nighthawk result was malformed, contains no counter with name: ", counter_name));
+}
+
+const nighthawk::client::Statistic& FortioOutputFormatterImpl::getRequestResponseStatistic(
+    const nighthawk::client::Result& result) const {
+  for (auto const& nh_stat : result.statistics()) {
+    if (nh_stat.id() == "benchmark_http_client.request_to_response") {
+      return nh_stat;
+    }
+  }
+
+  throw NighthawkException("Nighthawk result was malformed, contains no "
+                           "'benchmark_http_client.request_to_response' statistic.");
+}
+
+std::string FortioOutputFormatterImpl::formatProto(const nighthawk::client::Output& output) const {
+  nighthawk::client::FortioResult fortio_output;
+
+  // TODO(nareddyt): Not needed but nice to have, displays in the UI
+  fortio_output.mutable_labels()->set_value("");
+  fortio_output.mutable_starttime()->set_seconds(output.timestamp().seconds());
+  fortio_output.mutable_requestedqps()->set_value(output.options().requests_per_second().value());
+  fortio_output.mutable_url()->set_value(output.options().uri().value());
+
+  // Actual and requested durations are the same
+  const auto& nh_duration = output.options().duration().seconds();
+  fortio_output.mutable_requestedduration()->set_seconds(nh_duration);
+  fortio_output.mutable_actualduration()->set_value(nh_duration);
+
+  // This displays as "connections" in the UI, not threads
+  fortio_output.mutable_numthreads()->set_value(output.options().connections().value());
```

Comment:
This is the requested setting, and not the effective setting, right? If I remember correctly, in Fortio this corresponds to the number of goroutines, right? Thinking about it, what we want here might be a bit more tricky; we must take into account Nighthawk's `--concurrency` flag, which acts as a multiplier. Furthermore, we have `--connections` when using h1, but with h2 that is ignored today as Envoy's h2 pool mostly uses a single connection. So we'd probably want the number of concurrent streams (governed by `--max-active-requests`) here, multiplied by the number specified in concurrency.

Reply:
Ok, I see why we shouldn't be using `--connections`.

However, I don't know if using `--concurrency` will work, since it's a string value that can be set to "auto". How can I determine the concurrency for auto values?

Reply:
I could get the number of CPU cores like the source code does, but its possible the output transform might run on a different machine with a different number of CPU cores, so it wouldn't be accurate.

Can we file a bug for this and address it in another PR, since this isn't a core value that is needed?

Reply:
Sounds good to me. Maybe also mention that a prerequisite to that is that we offer the effective concurrency in NH's proto, it sounds like we need something like that.

Reply:
#186